### PR TITLE
Clean up most of the analyzer info

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This turns a number of the code analyzers into warnings and fixes them. Most of the outstanding messages are fixed.

It wasn't particularly easy to break this up as the analyzers are pretty squidgy about reporting what is outstanding. Everything from this point should be much more focused.

We have significant weirdness with collection initialization syntax. It is ambiguous for most of our collection types and `AddRange`. Behavior varies between debug and release. Hopefully adding params of span overloads will address the ambiguity issue so that auto-fixes will actually create code that builds.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11050)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Apologies to other pending changes. Ping me with any questions on conflict resolution.

**Comment by paul1956** [comment]
A lot of it is style Using imports and just function calls vs Function calls the include part or all of the name space. I prefer long imports with short statements, VB has a mixture of both (even in same file) so you get a lot of suggestions. Reviewer seem to have different opinions and sometimes prefer VB import aliases. Since VB is already verbose I prefer things that make lines shorter and more readable.Also I have split editor.config with specific preferences for VB and C#, Klaus prefers longer more explicit statements for example. Dim x as Integer = 6‘ first is preferred for VB but C# is opposite Dim y as BooleanIf x = 3 then    y = TrueElse     y FalseEnd If‘ Vs thisDim y as Boolean = x = 3His reasoning as I understand it is most people reading VB code will not understand some of the VB shortcuts and compiler generates same code. Sent from my iPhoneI apologize for any typos Siri might have made.(503) 803-6077On Mar 13, 2024, at 5:15 PM, Jeremy Kuhne ***@***.***> wrote:﻿
Apologies to other pending changes. Ping me with any questions on conflict resolution.

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ThemingScope.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public ThemingScope(bool useVisualStyles)
         }
     }
 
-    public void Dispose()
+    public readonly void Dispose()
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ThemingScope.cs:38]: This is weird... readonly void?

**Review by Tanya-Solyanik — APPROVED** [review]
I scrolled through this change, but I hope that code fixes work correctly, as I don't trust my eyes anymore.

**Comment by paul1956** [comment]
This is why, you write tests before this kind of automated change. Automated code fixes in Roslyn are very good but not perfect as several people have found out but usually they just cause errors vs silently do the wrong thing. Sent from my iPhoneI apologize for any typos Siri might have made.(503) 803-6077On Mar 13, 2024, at 8:07 PM, Tanya Solyanik ***@***.***> wrote:﻿
@Tanya-Solyanik approved this pull request.

I scrolled through this change, but I hope that code fixes work correctly, as I don't trust my eyes anymore.

In src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ArrayEditor.cs:
> @@ -44,7 +44,7 @@ protected override object[] GetItems(object editValue)
     /// </summary>
     protected override object SetItems(object editValue, object[] value)
     {
-        if (editValue is not null && !(editValue is Array))
+        if (editValue is not null and not Array)

This could be simplified as (editValue is not Array)



In src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.CollectionEditorCollectionForm.cs:
> @@ -997,9 +993,7 @@ private void UpButton_Click(object? sender, EventArgs e)
             {
                 SuspendEnabledUpdates();
                 int ti = _listBox.TopIndex;
-                object itemMove = _listBox.Items[index];
-                _listBox.Items[index] = _listBox.Items[index - 1];
-                _listBox.Items[index - 1] = itemMove;
+                (_listBox.Items[index - 1], _listBox.Items[index]) = (_listBox.Items[index], _listBox.Items[index - 1]);

What rule is this?Or was this a manual change?



In src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/CommandSet.cs:
>                              {
-                                if (componentNames is not null && idx < componentNames.Length)
+                                // If we couldn't find a property for this event, or of the property is read only, then
+                                // abort.
+                                //

remove this line



In src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ThemingScope.cs:
> @@ -35,7 +35,7 @@ public ThemingScope(bool useVisualStyles)
         }
     }
 
-    public void Dispose()
+    public readonly void Dispose()

This is weird... readonly void?

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.CollectionEditorCollectionForm.cs [hunk 2]
```diff
@@ -997,9 +993,7 @@ private void UpButton_Click(object? sender, EventArgs e)
             {
                 SuspendEnabledUpdates();
                 int ti = _listBox.TopIndex;
-                object itemMove = _listBox.Items[index];
-                _listBox.Items[index] = _listBox.Items[index - 1];
-                _listBox.Items[index - 1] = itemMove;
+                (_listBox.Items[index - 1], _listBox.Items[index]) = (_listBox.Items[index], _listBox.Items[index - 1]);
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.CollectionEditorCollectionForm.cs:996]: Use tuples to swap values

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ThemingScope.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ThemingScope.cs:38]: 'readonly' in this location states that the method will not change the state of the object. This allows C# to create more optimized code (avoiding defensive struct copies for example).

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ArrayEditor.cs [hunk 3]
```diff
@@ -44,7 +44,7 @@ protected override object[] GetItems(object editValue)
     /// </summary>
     protected override object SetItems(object editValue, object[] value)
     {
-        if (editValue is not null && !(editValue is Array))
+        if (editValue is not null and not Array)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ArrayEditor.cs:47]: Yeah, I caught some of these. I'll catch it in another PR.

**Comment by JeremyKuhne** [comment]
@paul1956 We can iterate on VB style with @KlausLoeffelmann.

While there is some risk to making changes of this magnitude the benefits to the long-term health of the codebase are significant. We had over 6000 "info" messages in the repo (probably more than 10K before I fixed a number of others). It was difficult to see whether or not you were introducing additional ones when modifying/adding code. It also had a significant impact on the perf in VS. This will also reduce contribution overhead as we won't have to spend as much time manually pointing out style guidelines. :)

**Comment by paul1956** [comment]
I am a huge proponent of getting to 0 “info” messages, when I managed software development in the stone age it was a requirement for every project.I have written many code fix’s beyond what Roslyn offers like forcing VB Event names to match event handlers, which rename doesn’t do and fix accidentally deletion of handles clause when designer removed them (basically mirror fixes).I was just pointing out they have been know to break code as was pointed out to me by @kirsan31 when I tried to make DataVisualization “info” message free.Sent from my iPhoneI apologize for any typos Siri might have made.(503) 803-6077On Mar 14, 2024, at 8:17 AM, Jeremy Kuhne ***@***.***> wrote:﻿
@paul1956 We can iterate on VB style with @KlausLoeffelmann.
While there is some risk to making changes of this magnitude the benefits to the long-term health of the codebase are significant. We had over 6000 "info" messages in the repo (probably more than 10K before I fixed a number of others). It was difficult to see whether or not you were introducing additional ones when modifying/adding code. It also had a significant impact on the perf in VS. This will also reduce contribution overhead as we won't have to spend as much time manually pointing out style guidelines. :)

**Pull request merged**